Rails4 upgrade

tom-lord opened the pull request:
All the rails 4 application compatibility stuff.

TODO: Update README.md

Commit:
Merge branch 'readme_grammar'

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Upgraded engine to run on rails 4

The engine should still be rails 3 compatable, IF you use the "strong
parameters" gem. I'll add something to the README about this, once I've
got it to work properly.

Commit:
Removed commented out code

Commit:
Removed commented out code

Commit:
Rails 3 compatability fix, and resolved a NilClass bug

Commit:
Rails 3 compatability fix, and resolved a NilClass bug

tom-lord commented on witch_doctor.gemspec:
```diff
@@ -18,14 +18,12 @@ Gem::Specification.new do |s|
   s.test_files = Dir["spec/**/*"]
 
-  s.add_dependency "rails", "~> 3.2"
+  s.add_development_dependency "rails", "~> 4.0.0"
```
This is the main line I'm really not sure about...

Maybe it should be something like:

```
s.add_dependency "rails", ">= 3.2"
s.add_development_dependency "rails", "~> 4.0.0"
```

?

equivalent replied:
hmm technically the requirements are 

```
s.add_dependency "rails", ">= 3.2"
s.add_dependency "strong_params"
s.add_development_dependency "rails", "~> 4.0.0"
```

so I'm lost here too :)  generaly this isn't issue for our app but becouse this is a public gem this may be an issue let's keep it for now like this

tom-lord replied:
Ah, but strong_params is a requirement if (and only if!) you are using rails 3 - so your code breaks it (I think) :)

What I have put here currently "works", but I think it may be missing the s.add_dependency - which is only really important for people using this as a "public gem"

equivalent commented on spec/dummy/config/config/secrets.yml:
```diff
@@ -0,0 +1,22 @@
+# if you're sharing your code publicly.
+
+development:
+  secret_key_base: 8a97515bd4f99f3f1015539159e069c00cba636166257c367e42be26e7279650d1ce1954897ee86c8ac4991718565b984f2336b6511a2b1d37a524870e03bb56
```
???? is this the real token ?? ?

equivalent replied:
don't quite understand why you need this

equivalent replied:
oh ok this is Rails thing I see, not related to implementation, no  problem then :)

equivalent commented on spec/models/virus_scan_spec.rb:
```diff
@@ -5,7 +5,8 @@
 
     context 'after updating record with scan results' do
-      before { virus_scan.update_attributes({scan_result: 'Clean'}, as: :scan_update ) }
+      #before { virus_scan.update_attributes({scan_result: 'Clean'}, as: :scan_update ) }
```
remove pls the comment

equivalent commented:
@tom-lord  can you please fix `rake`  (it will create db if not exist; migrate database and run specs)

Commit:
Removed comment

Commit:
Removed comment

Commit:
Cleaned up dummy application / config

There was some duplication, due to some copying mistakes.

Also, fixed `rake spec`

Commit:
Merge with upstream

Commit:
Merge branch 'rails4_upgrade' of github.com:tom-lord/witch_doctor into rails4_upgrade

Conflicts:
	Gemfile.lock

tom-lord commented:
Urgh... that was a messy messy merge... Should be all good now though :+1:

equivalent commented:
:hamster:

equivalent commented:
can you pls write a spec for `VirusScanPermitter` ?

Commit:
Reverted NilClass patch

The correct solution is to ensure all resources in your application
(which existed prior to mounting this rails engine) have virus scans
scheduled.

Commit:
Permitter spec

tom-lord commented:
Spickity spec yo self before you wreck yo self

Commit:
Don't permit the scanned_at attribute

equivalent commented:
looks good dude :)
